x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -4499,6 +4500,31 @@ public void testReplaceSortByExpressionsWithStats() {
         as(aggregate.child(), EsRelation.class);
     }
 
+    /**
+     * Expects
+     * Limit[1000[INTEGER]]
+     * \_InlineStats[[emp_no % 2{r}#6],[COUNT(salary{f}#12) AS c, emp_no % 2{r}#6]]
+     *   \_Eval[[emp_no{f}#7 % 2[INTEGER] AS emp_no % 2]]
+     *     \_EsRelation[test][_meta_field{f}#13, emp_no{f}#7, first_name{f}#8, ge..]
+     */
+    public void testInlinestatsNestedExpressionsInGroups() {
```

Comment:
There's a couple of tests, `testNestedExpressionsInGroups` and below; could we place this next to them and/or generalize these tests to apply to `Stats` instead of `Aggregate`?